spec.bs

```diff
@@ -2471,19 +2502,52 @@ CORP violation report=] algorithm, as leaving it unfenced may cause a privacy le
   : <dfn for="source snapshot params">attribution reporting context origin</dfn>
   :: an [=origin=].
 
+  : <dfn for="source snapshot params">automatic beacons allowed</dfn>
+  :: an [=boolean=].
+
+  : <dfn for="source snapshot params">automatic beacon data map</dfn>
+  :: a [=map=] whose [=map/keys=] are [=fencedframetype/automatic beacon event type=]s and whose
+     [=map/values=] are null or an [=fencedframetype/automatic beacon data=]
+
   Note: The [=source snapshot params/initiator fenced frame config instance=] is the [=fenced frame
   config instance=] that's loaded into a navigation initiator's [=browsing context=], if any exists.
-  It is used by the [=attempt to send an automatic beacon=] algorithm to retrieve the [=fenced frame
-  reporter/automatic beacon data map=] to be sent out if the <{fencedframe}>-initiated navigation
-  succeeds. The [=source snapshot params/target fenced frame config=] on the other hand, is the
-  non-[=instantiate a config|instantiated=] [=fenced frame config=] that will be loaded into a a
-  <{fencedframe}> element for navigations targeting fenced frames. These fields do not interact
-  *together* in any meaningful way.
+  It is used by the [=attempt to send an automatic beacon=] algorithm to compare [=origin=]s and
+  determine which {{FenceReportingDestination}}s to send beacons to, if the
+  <{fencedframe}>-initiated navigation succeeds. The [=source snapshot params/target fenced frame
+  config=] on the other hand, is the non-[=instantiate a config|instantiated=] [=fenced frame
+  config=] that will be loaded into a a <{fencedframe}> element for navigations targeting fenced
+  frames. These fields do not interact *together* in any meaningful way.
+</div>
+
+<div algorithm>
+  To <dfn>get the automatic beacon data mapping to use</dfn> given a {{Document}} |sourceDocument|:
+
+  1. Let |automatic beacon data map| be an empty [=map=] whose [=map/keys=] are [=fencedframetype/
+     automatic beacon event type=]s and whose [=map/values=] are null or an [=fencedframetype/
+     automatic beacon data=].
```

Comment:
Maybe we could just make this a new empty "automatic beacon data map", since this seems to match exactly that definition's shape. It feels weird to initialize something as a "new" type that is only defined as a member type, but honestly it's probably fine and making this less redundant is probably worth it.